sn_networking/src/driver.rs

```diff
@@ -616,6 +638,35 @@ impl SwarmDriver {
     // ---------- Crate helpers -------------------
     // --------------------------------------------
 
+    /// Return the record we hold that is farthest and relevant to us
+    fn get_farthest_relevant_record(&mut self, closest_k_peers: &Vec<PeerId>) -> Option<Distance> {
+        let mut farthest_distance = None;
+        #[allow(clippy::mutable_key_type)]
+        let locally_stored_keys = self
+            .swarm
+            .behaviour_mut()
+            .kademlia
+            .store_mut()
+            .record_addresses_ref();
+        let our_address = NetworkAddress::from_peer(self.self_peer_id);
+
+        for (_key, (address, _r_type)) in locally_stored_keys.iter() {
+            if Self::is_in_close_range(&self.self_peer_id, address, closest_k_peers) {
```

Comment:
instead of calculate closeness against entire stored key_set, which is costive and perhaps un-necessary as well.
can we use a simpler calculation say:
the distance range shall just be the distance from us to the 20th closest to us ?
this may let us achieve the same purpose with certain level of margin to allow glitch ?

Reply:
I had wondered that, but then is the not _not_ the CLOSE_RANGE which is what's stored at the closest 5 nodes 🤔 ? 

(or whats the line between 5 and 6 and our resonsibility... I know what you mean, but it's tricky... I'm happy to look for a better way here though)